mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -2511,6 +2511,41 @@ static bool UpdateAssemblyVersionForAssemblyInfo (XmlElement e, XmlElement root,
 		"System.Runtime.CompilerServices.DynamicAttribute",
 	};
 
+	private IEnumerable<char> FilterSpecialChars (string value)
```

Comment:
Because the incoming string isn't simply `"\0"`, it's the whole attribute value that gets written to the XML (`"System.ComponentModel.DefaultValue('\0')"`). So I'm just enumerating through each char in the string, and when it encounters one of the special chars, I return two chars in its place. `yield return` is a convenient way of doing that :)